build/build.sh

```diff
@@ -306,19 +306,36 @@ function InstallDotNetCli {
 }
 
 function InstallRepoToolset {
+  CreateDirectory "$RepoToolsetDir"
   ReadJson "$RepoRoot/global.json" "RoslynTools.RepoToolset"
   RepoToolsetVersion=$readjsonvalue
-  RepoToolsetDir="$NuGetPackageRoot/roslyntools.repotoolset/$RepoToolsetVersion/tools"
-  RepoToolsetBuildProj="$RepoToolsetDir/Build.proj"
+  RepoToolsetLocationFile="$RepoToolsetDir/$RepoToolsetVersion.txt"
+
+  if [[ -a "$RepoToolsetLocationFile" ]]; then
+    local path=`cat $RepoToolsetLocationFile`
+    if [[ -a "$path" ]]; then
+      RepoToolsetBuildProj=$path
```

Comment:
So the logic is "if repotools appears to have already been restored (has a path written to a text file) then just return that path, otherwise write a project and have it write the path, "restoring" the SDK automatically during that"?